src/utils/portfolioCalculations/getCurrentPortfolio.ts

```diff
@@ -0,0 +1,103 @@
+import {
+  getSortedPurchaseTransactions,
+  getSortedSaleTransactions,
+} from './getSortedTransactions';
+import {
+  calculationParams,
+  CurrentPortfolioItem,
+  CurrentPortfolioTransaction,
+} from '../../types/calculations';
+
+export const getCurrentPortfolio = ({
+  existingTransactions,
+  savedTransactions,
+}: calculationParams) => {
+  const transactions = [...existingTransactions, ...savedTransactions];
+
+  const purchaseTransactions: CurrentPortfolioTransaction[] =
+    getSortedPurchaseTransactions(transactions).map((tx) => ({
+      ...tx,
+      remainingNumberOfStocksOwned: tx.numberOfStocks,
+    }));
+  const saleTransactions = getSortedSaleTransactions(transactions).map(
+    (tx) => ({
+      ...tx,
+      remainingNumberOfStocksToSell: tx.numberOfStocks,
+    }),
+  );
+
+  saleTransactions.forEach((saleTx) => {
+    let remainingNumberOfStocksToSell = saleTx.remainingNumberOfStocksToSell;
+
+    purchaseTransactions.forEach((purchaseTx) => {
+      if (remainingNumberOfStocksToSell <= 0) return;
+      if (purchaseTx.holding.holdingSymbol !== saleTx.holding.holdingSymbol)
+        return;
+
+      const stocksToSell = Math.min(
+        remainingNumberOfStocksToSell,
+        purchaseTx.remainingNumberOfStocksOwned,
+      );
+
+      purchaseTx.remainingNumberOfStocksOwned -= stocksToSell;
```

Comment:
You should never mutate an objects property. This can easily lead to bugs. Instead you should calculate the new value, return it, then assign it to the object witch copying all the old properties and inserting a new value where it's needed. 

Both of the `forEach` should be converted to reduce, then probably it will be more obvious what has to be done.